### PR TITLE
Added AttributeError when trying to set mask on non-masked table

### LINKED ISSUE #3505: Setting mask on non-masked column should raise error
In the following example, the table is not masked, so I feel that for safety we should raise an ``AttributeError`` for the last statement:

```
In [2]: t = Table()

In [3]: t['a'] = [1,2,3]

In [4]: t['a'].mask = [True, False, True]
```

cc @taldcroft @mhvk

### PR DESCRIPTION
**Pull request by mvnnn**
fixes #3505

### CODE REVIEW

**Comment by mvnnn** [comment]
ping @taldcroft

File: astropy/table/column.py [hunk 1]
```diff
@@ -706,6 +706,11 @@ def __new__(cls, data=None, name=None,
                                           copy=copy, copy_indices=copy_indices)
         return self
 
+    def __setattr__(self, item, value):
+        if not isinstance(self, MaskedColumn) and item == "mask":
+            raise AttributeError("Cannot set mask value to a Column in non-masked Table")
```
Comment (taldcroft) [inline astropy/table/column.py]: Use lower case to start the error message, i.e. `cannot set ..`.  We are not consistent about this but new code should follow that convention.

File: astropy/table/tests/test_column.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 from ...tests.helper import pytest, assert_follows_unicode_guidelines
 from ... import table
+from ..table import Table
```
Comment (taldcroft) [inline astropy/table/tests/test_column.py]: Use use the existing `table` import above with `table.Table`.  Most of the time one should minimize changes and follow the existing convention in a module.

**Comment by taldcroft** [comment]
@mvnnn - please add an entry to the new features for 1.2 and address my minor comments, then this will be good.  Thanks!

**Comment by taldcroft** [comment]
(I meant new features in CHANGES.rst).

**Comment by taldcroft** [comment]
Thanks @mvnnn !

**Pull request merged**